### PR TITLE
Updates docker publishing

### LINKED ISSUE #4336: Change the flow of publishing docker build to update latest after publishing a release
## Abstract

Users might depend on the `latest` tag to update their containers, e.g. auto-updating using watchtower. 

With the current release flow, `latest` is updated as soon as the build finishes.
Sometimes this might be too early as last moment bugs might be detected, and users might update to broken builds.

We should only update the latest tag on release.

### TODO
- [ ] Keep a tag updated on build to make testing possible. 
- [ ] See if there is a way to re-tag an existing image on docker without re-build
- [ ] Add a github action??? that updates latest on release

### LINKED ISSUE #4535: Update docker Tag to depend on repo env variable
## Problem Definition

Accidentally update latest testing release in the fork.

GitHub actions provides an env variable that gives the owner/repo string.

### PR DESCRIPTION
**Pull request by kelsos**
Closes #4336 
Closes #4535

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: .github/workflows/rotki_docker_publish.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Cache regctl
+        uses: actions/cache@v3
+        with:
+          path: regctl-linux-amd64
+          key: ${{ runner.os }}-regctl-${{ env.REGCTL }}
+      - name: Publish ${{ github.repository }}:latest from ${{ steps.rotki_version.outputs.version }}
+        run: |
+          if [[ ! -f "regctl-linux-amd64" ]]; then
+            echo "downloading regctl"
+            curl -LO "https://github.com/regclient/regclient/releases/download/${{ env.REGCTL }}/regctl-linux-amd64"
```
Comment (yabirgb) [inline .github/workflows/rotki_docker_publish.yaml:36]: Is a bit ugly that you have to use this to install regctl but seems to be the official way

**Review by yabirgb — APPROVED** [review]
Looks good and will solve the problem we have with latest

File: .github/workflows/rotki_docker_publish.yaml (hunk 1 above)
Comment (kelsos) [reply .github/workflows/rotki_docker_publish.yaml:36]: I am not even sure if it is in the official repos, but it seemed the fastest way to get only the specific utility. We don't need everything else in `regclient`.

Also basically every action that uses software that is not pre-installed in the runner pretty much downloads the software from somewhere.

Comment (yabirgb) [reply .github/workflows/rotki_docker_publish.yaml:36]: Yes, in the repo they also have this as installation method

**Pull request merged**